src/commands/remoteDebug/getRemoteDebugLanguage.ts

```diff
@@ -6,15 +6,19 @@
 import { SiteConfig } from '@azure/arm-appservice';
 import { RemoteDebugLanguage } from '@microsoft/vscode-azext-azureappservice';
 
-export function getRemoteDebugLanguage(siteConfig: SiteConfig): RemoteDebugLanguage {
+export function getRemoteDebugLanguage(siteConfig: SiteConfig, appServicePlan?: string): RemoteDebugLanguage {
     // Read siteConfig.linuxFxVersion to determine debugging support
     //   If the Function App is running on Windows, it will be empty
-    //   If the Function App is running on Linux consumption, it will be empty
+    //   If the Function App is running on Linux consumption plan remote debugging is not supported
     //   If the Function App is running on a Linux App Service plan, it will contain Docker registry information, e.g. "DOCKER|repo.azurecr.io/image:tag"
-    //      The blessed Node.js Docker image will be something like "DOCKER|mcr.microsoft.com/azure-functions/node:2.0-node8-appservice"
-    if (siteConfig.linuxFxVersion && siteConfig.linuxFxVersion.startsWith('DOCKER|mcr.microsoft.com/azure-functions/node')) {
-        return RemoteDebugLanguage.Node;
+    //      or it will contain language information, e.g. "Node|14"
+    const isNotConsumption: boolean = appServicePlan?.toLowerCase() !== 'y';
+    if (isNotConsumption) {
+        if (siteConfig.linuxFxVersion && (/^(DOCKER\|(mcr\.microsoft\.com)\/(azure-functions)\/(node)|Node\|)/i).test(siteConfig.linuxFxVersion)) {
```

Comment:
That wasn't working for me in regex101, but you don't need to group the different parts of the DOCKER string in parens anyway. I think that this should work since we're expecting an exact match on the DOCKER string.

```suggestion
        if (siteConfig.linuxFxVersion && (/(DOCKER\|mcr\.microsoft\.com\/azure-functions\/node)|(Node\|)/i).test(siteConfig.linuxFxVersion)) {
```